### PR TITLE
Updated breaking changes section

### LINKED ISSUE #5574: [Core] Removing a contracted model now constitutes a breaking change
Removing a contracted model now constitutes a breaking change: 
* if the model is versioned -> error
* if the model is unversioned -> warning

A model is considered 'removed' if it is deleted, renamed, or disabled (via config: enabled)

Updating the docs here would make sense! https://docs.getdbt.com/docs/collaborate/govern/model-contracts#how-are-breaking-changes-handled

Issue: https://github.com/dbt-labs/dbt-core/issues/10116
Pull request: https://github.com/dbt-labs/dbt-core/pull/10221

### PR DESCRIPTION
**Pull request by nataliefiann**
Closes https://github.com/dbt-labs/docs.getdbt.com/issues/5574

Following this Git issue: https://github.com/orgs/dbt-labs/projects/14/views/35?pane=issue&itemId=65030468, i've created this PR regarding the updated breaking changes

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [ ] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 1]
```diff
@@ -206,9 +206,13 @@ At the same time, for models with many columns, we understand that this can mean
 When comparing to a previous project state, dbt will look for breaking changes that could impact downstream consumers. If breaking changes are detected, dbt will present a contract error. 
 
 Breaking changes include:
-- Removing an existing column
-- Changing the `data_type` of an existing column
-- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher)
+- Removing an existing column.
+- Changing the `data_type` of an existing column.
+- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher).
+- Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled (by config: enabled):
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]: Suggest leaving that example out as the "enabled" language might be confusing.
```suggestion
- Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled:
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 2]
```diff
@@ -206,9 +206,13 @@ At the same time, for models with many columns, we understand that this can mean
...
+  - if the model is versioned, dbt will throw an error.
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
  - If the model is versioned, dbt will throw an error.
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 3]
```diff
@@ -206,9 +206,13 @@ At the same time, for models with many columns, we understand that this can mean
...
+  - if the model is unversioned, dbt will raise a warning.
```
Comment (matthewshaver) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
  - If the model is unversioned, dbt will raise a warning.
```

**Review by matthewshaver — APPROVED** [review]

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 4]
```diff
@@ -206,9 +206,13 @@ At the same time, for models with many columns, we understand that this can mean
...
 
 Breaking changes include:
-- Removing an existing column
-- Changing the `data_type` of an existing column
-- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher)
+- Removing an existing column.
+- Changing the `data_type` of an existing column.
+- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher).
+- Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled:
+  - If the model is versioned, dbt will throw an error.
```
Comment (runleonarun) [inline website/docs/docs/collaborate/govern/model-contracts.md]: Hey @nataliefiann I had spun up a PR for this locally so happen to look at this and suggest these edits:
```suggestion
  - When you remove a versioned model, you will receive an error.
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 5]
```diff
@@ -206,9 +206,13 @@ At the same time, for models with many columns, we understand that this can mean
...
+  - If the model is unversioned, dbt will raise a warning.
```
Comment (runleonarun) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
  - When you remove an unversioned contract, you will receive a warning.
```

**Review by runleonarun — CHANGES_REQUESTED** [review]
Just suggesting to add some clarity to the warning/error here!

Also, should we revise the wording here to match your wording? https://docs.getdbt.com/reference/resource-properties/versions#detecting-breaking-changes

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 6]
```diff
@@ -206,9 +206,13 @@ At the same time, for models with many columns, we understand that this can mean
 When comparing to a previous project state, dbt will look for breaking changes that could impact downstream consumers. If breaking changes are detected, dbt will present a contract error. 
 
 Breaking changes include:
-- Removing an existing column
-- Changing the `data_type` of an existing column
-- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher)
+- Removing an existing column.
+- Changing the `data_type` of an existing column.
+- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher).
+- Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled:
```
Comment (nataliefiann) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion

<VersionBlock firstVersion="1.9">

- Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled:
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 7]
```diff
@@ -206,9 +206,16 @@ At the same time, for models with many columns, we understand that this can mean
...
+- Removing an existing column.
+- Changing the `data_type` of an existing column.
+- Removing or modifying one of the `constraints` on an existing column (dbt v1.6 or higher).
+
+<VersionBlock firstVersion="1.9">
+
+- Removing a contracted model constitutes a breaking change. A model is considered 'removed' if it is deleted, renamed, or disabled:
+  - When you remove a versioned model, you will receive an error.
+  - When you remove an unversioned contract, you will receive a warning.
+
```
Comment (nataliefiann) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
</VersionBlock>
```

**Comment by nataliefiann** [comment]
Closed off as Core project has been paused

**Review by runleonarun — APPROVED** [review]
LGTM! We can follow up on my comment

**Comment by dbeatty10** [comment]
> Just suggesting to add some clarity to the warning/error here!
> 
> Also, should we revise the wording here to match your wording? https://docs.getdbt.com/reference/resource-properties/versions#detecting-breaking-changes

Good catch! The relevant code is [here](https://github.com/dbt-labs/docs.getdbt.com/blob/ce4a8475fd65622e66ca34f77b5a0b2dd83e192f/website/docs/reference/resource-properties/versions.md?plain=1#L76-L81).

I've opened up a docs issue in https://github.com/dbt-labs/docs.getdbt.com/issues/6198 to see if these can be abstracted into a common snippet.

**Review by dbeatty10 — APPROVED** [review]

**Pull request merged**